### PR TITLE
feature/mx-1459 Expand documentation for identifiers and provenance

### PR DESCRIPTION
**Pull request by cutoffthetop**

### CODE REVIEW

File: mex/common/models/extracted_data.py [hunk 1]
```diff
@@ -35,23 +61,42 @@ def __str__(self) -> str:
...
+        because otherwise we cannot reliably determine the origin of this item.
+
+        Next we query the configured `IdentityProvider` to determine whether this item
+        already has an `identifier` and `stableTargetId`.
+        If not, we let the identity provider generate new identifiers.
+
+        If an `identifier` has been passed to the constructor, we check that it matches
+        with what we got from the identity provider, because we don't allow any system
+        to change the association from `identifierInPrimarySource` and
+        `hadPrimarySource` to `identifier`.
```
Comment (erichesse) [inline mex/common/models/extracted_data.py:95]: As I understand it, the identifier provision and matching happens entirely under the hood and we never need to touch it on the extractor level. At least for me this is very useful to know.

File: mex/common/models/extracted_data.py [hunk 2]
```diff
@@ -35,23 +61,42 @@ def __str__(self) -> str:
...
+
+        If a `stableTargetID` has been passed to the constructor, we use that as the
+        new value, because changes to the stable target ID are generally allowed.
+        For example, when matching to extracted items that describe the same thing.
```
Comment (erichesse) [inline mex/common/models/extracted_data.py:99]: After reading this part I am confused again. What is exactly the purpose of the "stable" target ID, when changes are generally allowed?

**Review by erichesse — COMMENTED** [review]
Thank you, looks very good. Two issues left to discuss.

File: mex/common/models/extracted_data.py (hunk 2 above)
Comment (cutoffthetop) [reply mex/common/models/extracted_data.py]: tried to address that here: https://github.com/robert-koch-institut/mex-common/pull/36/commits/6978ccd6f76231056b1d163d294379cb18b3d11c#diff-aa31d93942aa11a22383bc209495b8c58de08471a2e690d69a0a05770f5b6817R170

File: mex/common/models/extracted_data.py (hunk 1 above)
Comment (cutoffthetop) [reply mex/common/models/extracted_data.py]: hopefully clearer like this: https://github.com/robert-koch-institut/mex-common/pull/36/commits/6978ccd6f76231056b1d163d294379cb18b3d11c#diff-aecf8b33ed45196c6b75dd26ebf6fbc38df360a838f53491ef03a42a4744965bR87

**Review by erichesse — APPROVED** [review]
looks good, thanks for the additions

**Pull request merged**